### PR TITLE
Cleanup NEG Metrics

### PR DESCRIPTION
**Pull request by swetharepakula**
* Move metrics around to be easier to find
* Have package separation between metrics. Mainly to reduce dependencies on the main metrics package so that can be used to publish metrics in negtypes (I will be adding metrics as part of https://github.com/kubernetes/ingress-gce/pull/2150)

### CODE REVIEW

File: pkg/neg/controller.go [hunk 1]
```diff
@@ -40,7 +40,8 @@ import (
 	"k8s.io/ingress-gce/pkg/flags"
 	usageMetrics "k8s.io/ingress-gce/pkg/metrics"
 	"k8s.io/ingress-gce/pkg/neg/metrics"
-	syncMetrics "k8s.io/ingress-gce/pkg/neg/metrics"
+	"k8s.io/ingress-gce/pkg/neg/metrics/metricscollector"
```
Comment (sawsa307) [inline pkg/neg/controller.go:43]: I think these two imports are the same?

Comment (swetharepakula) [reply pkg/neg/controller.go]: thanks for catching that!

File: pkg/neg/metrics/metricscollector/metrics_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package metrics
+package metricscollector
```
Comment (gauravkghildiyal) [inline pkg/neg/metrics/metricscollector/metrics_test.go:17]: We can likely get rid of this file as the code would make sense within the neg_metrics_collector_test.go file.

File: pkg/neg/metrics/metricscollector/neg_metrics_collector.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package metrics
+package metricscollector
```
Comment (gauravkghildiyal) [inline pkg/neg/metrics/metricscollector/neg_metrics_collector.go:17]: Nit: Maybe we can change the file name to the match the package name -- metrics_collector.go

File: pkg/neg/metrics/metricscollector/neg_metrics_collector.go [hunk 4]
```diff
@@ -29,6 +29,19 @@ import (
 	"k8s.io/utils/clock"
 )
 
+var register sync.Once
+
+// RegisterSyncerMetrics registers syncer related metrics
+func RegisterMetrics() {
```
Comment (gauravkghildiyal) [inline pkg/neg/metrics/metricscollector/neg_metrics_collector.go:35]: I think we missed registering some DualStack metrics like:

DualStackMigrationFinishedDurations
DualStackMigrationLongestUnfinishedDuration
...

Maybe moving the RegisterMetrics() within the metricscollector/metrics.go file (where the metrics are defined) would help avoid this.

**Review by gauravkghildiyal — COMMENTED** [review]
Super helpful PR! Love the removal of that pile of unnecessary constants :)

File: pkg/neg/metrics/metricscollector/neg_metrics_collector.go (hunk 4 above)
Comment (swetharepakula) [reply pkg/neg/metrics/metricscollector/neg_metrics_collector.go]: Thanks for catching that!

File: pkg/neg/metrics/metricscollector/metrics_test.go (hunk 2 above)
Comment (swetharepakula) [reply pkg/neg/metrics/metricscollector/metrics_test.go]: Yes, the test fits with the metrics collector.

File: pkg/neg/metrics/metricscollector/neg_metrics_collector.go (hunk 3 above)
Comment (swetharepakula) [reply pkg/neg/metrics/metricscollector/neg_metrics_collector.go]: good point!

**Comment by gauravkghildiyal** [comment]
Thanks!

/lgtm

**Comment by swetharepakula** [comment]
/retest

verification failed in downloading go dependencies for the linter and not related to the PR

**Pull request merged**